src/tree-view/index.tsx

```diff
@@ -0,0 +1,28 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+
+import { getBaseProps } from '../internal/base-component';
+import useBaseComponent from '../internal/hooks/use-base-component';
+import { applyDisplayName } from '../internal/utils/apply-display-name';
+import { getExternalProps } from '../internal/utils/external-props';
+import { TreeViewProps } from './interfaces';
+import InternalTreeView from './internal';
+
+export { TreeViewProps };
+
+const TreeView = <T,>({ items, ...rest }: TreeViewProps<T>) => {
+  const baseProps = getBaseProps(rest);
+  const baseComponentProps = useBaseComponent('TreeView', {
+    props: {},
+    metadata: {
+      itemsCount: items?.length,
```

Comment:
nit: the items array is always defined as per TS declaration, so no need to use optional chaining here. Also, what do we need the items count for? This will likely change during the component lifetime, so can be zero at first, and then change to non-zero, after data is loaded.